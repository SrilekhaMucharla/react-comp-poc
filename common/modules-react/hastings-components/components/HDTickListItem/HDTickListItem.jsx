import React from 'react';

import PropTypes from 'prop-types';
import HDLabelRefactor from '../HDLabelRefactor/HDLabelRefactor';

import tickIcon from './icons/icons-tick.svg';
import crossIcon from './icons/icons-cross.svg';

const HDTickListItem = ({ children, title, selected }) => (
    <div className="tick-list-item-container">
        <div>{(selected) ? tickIcon() : crossIcon()}</div>
        <div className="tick-list-item__content">
            <HDLabelRefactor Tag="h5" text={title} />
            {children && <p className="tick-list-item__description">{children}</p>}
        </div>
    </div>
);

HDTickListItem.propTypes = {
    children: PropTypes.node,
    title: PropTypes.string.isRequired,
    selected: PropTypes.bool,
};
HDTickListItem.defaultProps = {
    children: null,
    selected: false,
};

export default HDTickListItem;
