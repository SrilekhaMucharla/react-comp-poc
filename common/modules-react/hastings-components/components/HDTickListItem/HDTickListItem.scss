.tick-list-item-container {
    display: flex;
    margin-top: 20px;
}

.tick-list-item__content {
    padding-left: 12px;
    text-align: left;
    margin-top: 3px;
}

.tick-list-item__description {
    margin-bottom: 0;
    margin-top: 8px;
}
