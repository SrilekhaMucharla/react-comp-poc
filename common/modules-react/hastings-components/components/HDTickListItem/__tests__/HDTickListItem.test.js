import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import HDTickListItem from '../HDTickListItem';

Enzyme.configure({ adapter: new Adapter() });

describe('<HDTickListItem />', () => {
    it('render component', () => {
        const wrapper = shallow(<HDTickListItem title="test" />);
        expect(wrapper).toMatchSnapshot();
    });
    it('render component with children', () => {
        const wrapper = shallow(<HDTickListItem title="test"><div /></HDTickListItem>);
        expect(wrapper).toMatchSnapshot();
    });
});
