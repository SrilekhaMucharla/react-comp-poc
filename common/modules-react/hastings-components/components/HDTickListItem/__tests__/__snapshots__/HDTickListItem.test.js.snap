// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDTickListItem /> render component 1`] = `
<div
  className="tick-list-item-container"
>
  <div>
    <svg
      fill="none"
      height="18"
      viewBox="0 0 32 32"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="16"
        cy="16"
        fill="#E40034"
        r="15"
        stroke="#fff"
        strokeWidth="2"
      />
      <path
        d="M17.678 16.00l5.158-5.108 1.051-1.051c.15-.15.15-.401 0-.601l-1.101-1.102c-.2-.15-.451-.15-.601 0l-6.16 6.21-6.21-6.21c-.15-.15-.4-.15-.6 0l-1.102 1.102c-.15.2-.15.45 0 .6l6.21 6.16-6.21 6.21c-.15.15-.15.4 0 .6l1.101 1.103c.2.15.451.15.601 0l6.21-6.21 5.108 5.158 1.052 1.052c.15.15.4.15.6 0l1.102-1.102c.15-.2.15-.45 0-.601l-6.21-6.21z"
        fill="#fff"
      />
    </svg>
  </div>
  <div
    className="tick-list-item__content"
  >
    <HDLabelRefactor
      Tag="h5"
      adjustImagePosition={true}
      className=""
      text="test"
    />
  </div>
</div>
`;

exports[`<HDTickListItem /> render component with children 1`] = `
<div
  className="tick-list-item-container"
>
  <div>
    <svg
      fill="none"
      height="18"
      viewBox="0 0 32 32"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="16"
        cy="16"
        fill="#E40034"
        r="15"
        stroke="#fff"
        strokeWidth="2"
      />
      <path
        d="M17.678 16.00l5.158-5.108 1.051-1.051c.15-.15.15-.401 0-.601l-1.101-1.102c-.2-.15-.451-.15-.601 0l-6.16 6.21-6.21-6.21c-.15-.15-.4-.15-.6 0l-1.102 1.102c-.15.2-.15.45 0 .6l6.21 6.16-6.21 6.21c-.15.15-.15.4 0 .6l1.101 1.103c.2.15.451.15.601 0l6.21-6.21 5.108 5.158 1.052 1.052c.15.15.4.15.6 0l1.102-1.102c.15-.2.15-.45 0-.601l-6.21-6.21z"
        fill="#fff"
      />
    </svg>
  </div>
  <div
    className="tick-list-item__content"
  >
    <HDLabelRefactor
      Tag="h5"
      adjustImagePosition={true}
      className=""
      text="test"
    />
    <p
      className="tick-list-item__description"
    >
      <div />
    </p>
  </div>
</div>
`;
